transport_nantes/topicblog/models.py

```diff
@@ -310,6 +315,26 @@ def get_participating_field_names(self) -> set:
                     continue
         return fields
 
+    def get_transactional_send_record_class(self) \
+            -> Type['SendRecordTransactional']:
+        """Return appropriate class for sending transactional emails.
+
+        Return the class of the transactional SendRecord object associated
+        with this topicBlog Object.
+        """
+        if self.transactional_send_record_class is None:
+            raise NotImplementedError
+        try:
+            send_record_class = \
+                apps.get_model('topicblog',
```

Comment:
Maybe it's worth a comment why we chose to use strings and then look them up in the global dictionary rather than letting the derived class just do its thing?

Reply:
We do the operation in that sense : We're given a TB object (be it Email or press) and from here, need to figure out its corresponding Transactional send record. 
Given that, I can't use the derived class because this is precisely what I'm looking for.

Reply:
This comment in the PR is notably not a comment in the code.

That said, this is a function called `get_transactional_send_record_class()` that lives in `TopicBlogObjectBase`.  So this presents two questions to me:

1.  Why does `TopicBlogObjectBase` even know what a send record is?  That knowledge is injected by sendable mixins in derived classes.  I think you're taking too free advantage of the fact that you have total control over the code and so putting stuff where it's most convenient at the moment rather than where it is most maintainable later.

2.  If this were the way to do it, derived classes could have methods that return whatever is right for them (possibly calling `super()`) and the base class' method would perhaps error because no such class is right for it.

3.  Isn't it the view that makes this association, not the model?

Reply:
I totally missed the fact that I can insert it inside the view. 
Welp, time to re-write, it would have been easier if I thought of that earlier ...